### PR TITLE
feat(): pg-query-stream typescript

### PR DESCRIPTION
**Pull request by chyzwar**
New version should emit types compatible with
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/47e9cf6d8947e46a82bb09bbcc3dd74a3ac0ffa1/types/pg-query-stream/index.d.ts

Let me know if this is going to right direction?

TODO:
- [x] Confirm readable option emitClose, is this is valid option? 
- [x] Add mocha setup for typescript
- [x] Convert tests to typescript
- [x] Ensure that emitted code is compatible with JS version

### CODE REVIEW

File: packages/pg-query-stream/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "strict": false,
```
Comment (brianc) [inline packages/pg-query-stream/tsconfig.json:6]: it will be nice to turn this to `true` one day, but this is fine for the initial port.

**Review by brianc — COMMENTED** [review]
this looks rad - lmk if you have any questions!

File: packages/pg-query-stream/tsconfig.json (hunk 1 above)
Comment (chyzwar) [reply packages/pg-query-stream/tsconfig.json:6]: I think this would be possible after pg-cursor is converted. I will look at pg-cursor next

**Comment by chyzwar** [comment]
I think this should be ready to review.

- Added solution project to compile pg-query-stream and pg-protocol (incremental and composite)
- Added files in pg-query-stream and pg-protocol to avoid publishing test files and buildInfo
- Remove emitClose from super call in QueryStream
- Remove unsued results from _read -> read
- Converted test to ts and changed var top consts
- Rename class from PgQueryStream to QueryStream

File: packages/pg-query-stream/src/index.ts [hunk 2]
```diff
@@ -1,11 +1,30 @@
...
+  handlePortalSuspended: Function
+  handleCommandComplete: Function
+  handleReadyForQuery: Function
+  handleError: Function
+  handleEmptyQuery: Function
+
+  public constructor(text: string, values?: any[], config: QueryStreamConfig = {}) {
     const { batchSize, highWaterMark = 100 } = config
-    // https://nodejs.org/api/stream.html#stream_new_stream_readable_options
-    super({ objectMode: true, emitClose: true, autoDestroy: true, highWaterMark: batchSize || highWaterMark })
```
Comment (brianc) [inline packages/pg-query-stream/src/index.ts:8]: curious: why remove `emitClose` here?

**Comment by brianc** [comment]
This looks solid - thanks a lot for doing this!  Sorry for the delay....have had a really horrible October - my doggie died. 😭 back in the saddle again now though.

**Comment by chyzwar** [comment]
emitClose is true by default and typescript was throwing error when It was trying to pass this option.

**Comment by chyzwar** [comment]
Sorry to hear about your dog.

**Comment by brianc** [comment]
@chyzwar sorry for the delay...life is crazy.  I'm down to merge this but there are a few merge conflicts.  Care to take a quick look at those & resolve them? Then i'll get this merged!  Thanks again! ❤️

**Comment by chyzwar** [comment]
this now has been rebased on master.

**Comment by brianc** [comment]
awesome! thanks! Will get this merged up today!

**Comment by brianc** [comment]
release pg@8.5.0

**Comment by vitaly-t** [comment]
Added issue #2412

**Pull request merged**